### ISSUE TITLE
Why doesn't Keras learn large gradient values?

### ISSUE THREAD

**Issue by ghostplant** [keras-team/keras#10847]
```sh
import keras
from keras.models import Sequential
from keras.layers import Dense
import numpy as np

num_classes = 2

model = Sequential()
model.add(Dense(512, activation='relu',
  kernel_initializer=keras.initializers.Constant(1.0),
  bias_initializer='zeros'))
model.add(Dense(512, activation='relu',
  kernel_initializer=keras.initializers.Constant(1.0),
  bias_initializer='zeros'))
model.add(Dense(2, activation='softmax',
  kernel_initializer=keras.initializers.Constant(1.0),
  bias_initializer='zeros'))

model.compile(loss='categorical_crossentropy',
              optimizer=keras.optimizers.SGD(lr=0.1, momentum=0.0,decay=0.0))

np.random.seed(10)
images = np.random.random([128, 224 * 224 * 3])
labels = np.random.random([128, 2])

model.fit(images, labels, batch_size=32, epochs=4)
```

The example above is a typical one what would cause `gradient explosion`.

We set all initial weights to be a large value and also using a large learning rate, so in theory, with the training step goes, the weights will become larger and larger and soon reach to `float(inf)`.

However, in Keras, the output is just:
```sh
Epoch 1/4
128/128 [==============================] - 3s 26ms/step - loss: 5.7850
Epoch 2/4
128/128 [==============================] - 0s 3ms/step - loss: 7.5418
Epoch 3/4
128/128 [==============================] - 0s 3ms/step - loss: 7.5418
Epoch 4/4
128/128 [==============================] - 0s 3ms/step - loss: 7.5418
```

It means that it gets weight updated only after the first step, which does learn gradient values whose values are around ~10.0^6f. But for the second step, it is expected to learn a 10^11f gradient but it just doesn't do any weights update. And with the help of `K.get_value(layers.get_weights[..])`, I just saw no gradients are learnt and all previous Dense weights keep unchanged, so does the 3rd and 4th training steps.

So what is the reason that makes Keras doesn't update large gradients (usually for those gradients larger than 10^8)? Is there default gradient clipping enabled?
